Paylike/Transaction.php

```diff
@@ -0,0 +1,109 @@
+<?php
+namespace Paylike;
+/**
+ * Class Transaction
+ * @package Paylike
+ * Handles transaction operations.
+ *
+ * @version    1.0.0
+ */
+class Transaction {
+
+	/**
+	 * @param $transaction_id
+	 *
+	 * @return int|mixed
+	 * Return the transaction data
+	 *
+	 */
+	public static function fetch( $transaction_id ) {
+		$adapter = Client::getAdapter();
+		if ( ! $adapter ) {
+			return - 1;
+		}
+
+		return $adapter->request( 'transactions/' . $transaction_id );
+
+	}
+
+
+	/**
+	 * @param $transaction_id
+	 * The authorization takes place via the client script,
+	 * so this actually returns the transaction data
+	 *
+	 * @return int|mixed
+	 */
+	public static function authorize( $transaction_id ) {
+		return self::fetch( $transaction_id );
+	}
+
+
+	/**
+	 * @param $transaction_id
+	 * @param $amount
+	 * @param $currency
+	 * @param string $description
+	 * Capture a transaction that has been authorized.
+	 * This also returns the transaction data.
+	 *
+	 * @return bool|int|mixed
+	 */
+	public static function capture( $transaction_id, $amount, $currency, $description = '' ) {
+		$adapter = Client::getAdapter();
+		if ( ! $adapter ) {
+			return - 1;
```

Comment:
Is returning `-1` the preferred way of communicating error in PHP? I would consider this type of error (missing configuration of adapter) as a "programmer error" thus I think it should throw a fatal error and halt the script unless someone wrapped it in a try-catch block further up.